### PR TITLE
strip whitespace for retryon

### PR DESCRIPTION
**Pull request by yxue**
Signed-off-by: Yan Xue <yxyan@google.com>

Description: strip whitespace for retryon
Risk Level: Low
Testing: Unit test
Docs Changes: N/A
Release Notes: N/A
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by yxue** [comment]
> Tagging as waiting while you sort out CI, but I wonder if it'd make sense to add this to splitToken. findToken has an argument for trimming whitespace and I think it would be generically useful.

Add `trim_whitespace` to the `splitToken` makes sense to me.

File: source/common/common/utility.cc [hunk 1]
```diff
@@ -322,18 +322,25 @@ absl::string_view StringUtil::cropLeft(absl::string_view source, absl::string_vi
...
+  std::vector<absl::string_view> result;
   if (keep_empty_string) {
-    return absl::StrSplit(source, absl::ByAnyChar(delimiters));
+    result = absl::StrSplit(source, absl::ByAnyChar(delimiters));
+  } else {
+    result = absl::StrSplit(source, absl::ByAnyChar(delimiters), absl::SkipEmpty());
+  }
+
+  if (trim_whitespace) {
+    for_each(result.begin(), result.end(), [](auto& v) { v = trim(v); });
```
Comment (alyssawilk) [inline source/common/common/utility.cc]: So if we have
"a, ,b"
and trim whitespace is true and keep empty string is false, we will end up returning an empty string, no?
Mind fixing and adding a unit test?

Comment (yxue) [reply source/common/common/utility.cc]: Done! Thanks for catching the bug! I added a test for this case.

**Comment by mattklein123** [comment]
> @mattklein123 we'd runtime guard this in-house but I think this is where we may be overly cautious so I wasn't inclined to ask for it here. SGTY?

I'm fine without a runtime guard but nor would I oppose it. Will defer to your judgement.

**Pull request merged**